lab_3_generate_by_ngrams/main.py

```diff
@@ -41,6 +43,23 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or text == '':
+            return None
+        clear_text = ''
+        for symbol in text.lower():
+            if not symbol.isalpha() or symbol.isdigit():
+                symbol = self._end_of_word_token
+            clear_text += symbol
+        i = 0
+        while i < len(clear_text) - 1:
```

Comment:
OK this is a very intricate algorithm... 

Any thoughts how it could be simpler? Try to do all that with just one iteration over symbols